addons/sale_project/models/project.py

```diff
@@ -281,8 +281,8 @@ def get_action(sol_id):
         } for sol_read in sols.with_context(with_price_unit=True).read(['display_name', 'product_uom_qty', 'qty_delivered', 'qty_invoiced', 'product_uom'])]
 
     def _get_sale_items_domain(self, additional_domain=None):
-        sale_orders = self._get_sale_order_items().sudo().order_id
-        domain = [('order_id', 'in', sale_orders.ids), ('is_downpayment', '=', False), ('state', 'in', ['sale', 'done']), ('display_type', '=', False)]
+        sale_order_lines = self._get_sale_order_items().sudo()
+        domain = [('id', 'in', sale_order_lines.ids), ('is_downpayment', '=', False), ('state', 'in', ['sale', 'done']), ('display_type', '=', False)]
```

Comment:
we can directly get the ids from SOLs that we want by calling the `_fetch_sale_order_item_ids` method instead. 🙂 
Btw, the `sudo` is not necessary if we just want the ids from the recordset. It is needed when we need to get another fields from the model and the current user has no access to that model. 🙂 

```suggestion
        sale_line_ids = self._fetch_sale_order_item_ids()
        domain = [('id', 'in', sale_line_ids), ('is_downpayment', '=', False), ('state', 'in', ['sale', 'done']), ('display_type', '=', False)]
```

Reply:
yeah I see, I committed your suggestion :smile:

Reply:
Thanks 🙂